fastapi/README.rst

```diff
@@ -1303,6 +1299,119 @@ for all the endpoints. You can do the same for a specific app by checking the
 'app' field of the 'fastapi.endpoint' record before registering your custom error
 handler.
 
+FastAPI addons directory structure
+==================================
+
+When you develop a new addon to expose an api with fastapi, it's a good practice
+to follow the same directory structure and naming convention for the files
+related to the api. It will help you to easily find the files related to the api
+and it will help the other developers to understand your code.
+
+Here is the directory structure that we recommend. It's based on practices that
+are used in the python community when developing a fastapi app.
+
+.. code-block::
+
+  .
+  ├── x_api_addon
+  │   ├── data
+  │   │   ├── ... .xml
+  │   ├── demo
+  │   │   ├── ... .xml
+  │   ├── i18n
+  │   │   ├── ... .po
+  │   ├── models
+  │   │   ├── __init__.py
+  │   │   ├── fastapi_endpoint.py  # your app
+  │   │   └── ... .py
+  │   └── routers
+  │   │   ├── __init__.py
+  │   │   ├── items.py
+  │   │   └── ... .py
+  │   ├── schemas
+  │   │   ├── __init__.py
+  │   │   ├── my_model.py  # pydantic model
+  │   │   └── ... .py
+  │   ├── security
+  │   │   ├── ... .xml
+  │   ├── views
+  │   │   ├── ... .xml
+  │   ├── __init__.py
+  │   ├── __manifest__.py
+  │   ├── dependencies.py  # custom dependencies
+  │   ├── error_handlers.py  # custom error handlers
+
+
+* The **'models'** directory contains the odoo models. When you define a new
+  app, as for the others addons, you will add your new model inheriting from
+  the **'fastapi.endpoint'** model in this directory.
+* The **'routers'** directory contains the fastapi routers. You will add your
+  new routers in this directory. Each route starting with the same prefix should
+  be grouped in the same file. For example, all the routes starting with
+  '/items' should be defined in the **'items.py'** file. The **'__init__.py'**
+  file in this directory is used to import all the routers defined in the
+  directory and create a global router that can be used in an app. In
+  each router file, you will define a router without prefix and omit the
+  in each route definition. The prefix will be added when the router is
+  imported in the **'__init__.py'** file. This will allows to easily use
+  the same router in different apps with different prefixes. For example,
+  in your **'items.py'** file, you will define a router like this:
+  .. code-block:: python
+
+    router = APIRouter()
+
+    router.get("/", response_model=List[Item])
+
+    def list_items(..):
```

Comment:
```suggestion
    @router.get("/items", response_model=List[Item])
    def list_items(..):
```

I'd prefer to read the last part of the route close to where it is implemented.

In `base_rest` that was a source of confusion for me.

Reply:
In such a case, it will never be possible to mount the router under another prefix (even if this will happen very rarely).  I improved the documentation on this point.

Reply:
Ok I simplified this point.